scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/GraphQLQueries.java

```diff
@@ -147,111 +148,181 @@ private JobInput getJobInput(List<String> statusNames, String user, List<Constra
                         }
-                    } catch (Exception e) {
-                        LOGGER.log(Level.SEVERE, "Error when parsing date filter \"" + value + "\"", e);
+                        break;
+                    }
+
+                    case STATE: {
+
+                        if (status == null)
+                            status = value.toUpperCase();
+                        else if (value.toUpperCase() != status)
+                            return input.jobName(RETURN_NOTHING_FILTER).build();
+                        break;
+                    }
+
+                    case PRIORITY: {
+
+                        if (priority == null)
+                            priority = value.toUpperCase();
+                        else if (value.toUpperCase() != priority)
+                            return input.jobName(RETURN_NOTHING_FILTER).build();
+                        break;
+                    }
+
+                    case USER: {
+
+                        if (constraint.getAction() == Action.EQUALS) {
+                            if (userFilter == null)
+                                userFilter = value;
+                            else if (value != userFilter)
+                                return input.jobName(RETURN_NOTHING_FILTER).build();
+                        } else if ((filter = getFilter(constraint, value)) != null)
+                            userFilter = filter;
+                        break;
+                    }
+
+                    case NAME: {
+
+                        if (constraint.getAction() == Action.EQUALS) {
+                            if (name == null)
+                                name = value;
+                            else if (value != name)
+                                return input.jobName(RETURN_NOTHING_FILTER).build();
+                        } else if ((filter = getFilter(constraint, value)) != null)
+                            name = filter;
+                        break;
+                    }
+
+                    case PROJECT_NAME: {
+
+                        if (constraint.getAction() == Action.EQUALS) {
+                            if (projectName == null)
+                                projectName = value;
+                            else if (value != projectName)
+                                return input.jobName(RETURN_NOTHING_FILTER).build();
+                        } else if ((filter = getFilter(constraint, value)) != null)
+                            projectName = filter;
+                        break;
+                    }
+
+                    case SUBMITTED_TIME: {
+
+                        try {
+
+                            dateInMs = ISODateTimeFormat.dateTimeParser().parseDateTime(value).toDate().getTime();
+
+                            switch (constraint.getAction()) {
+                                case GREATER_THAN_OR_EQUAL_TO:
+                                    if (afterSubmittedTime == -1 || dateInMs > afterSubmittedTime)
```

Comment:
Same as for ID. Change "afterSubmittedTime" and "beforeSubmittedTime" to String instead of long (as declared in jobInput class) and catch the exception when parsing the date.

Reply:
Good point.. i will homogenize the way I handle wrong format errors and handle it on the client side (graphically).